apis/v1alpha2/gateway_types.go

```diff
@@ -688,9 +691,9 @@ type ListenerConditionReason string
 
 const (
 	// This condition indicates that the controller was unable to resolve
-	// conflicting specification requirements for this Listener. If a
-	// Listener is conflicted, its network port should not be configured
-	// on any network elements.
+	// conflicting specification requirements for this Listener. If a Listener
+	// is conflicted, its network port should not be configured on any network
+	// elements. This condition MUST only be present when it is true.
```

Comment:
So its illegal to have `False`? or am I parsing this wrong?

Reply:
Yeah, I can clarify this, but the rationale was that `Conflicted=False` is a generally confusing state to have. I think it may be clearer to not have a condition in this case.

Reply:
IIRC, the general guidance for Conditions is that they are always published?

Reply:
That's a good point, somehow I'd thought that guidance was limited to positive conditions. Taking another look at the [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) suggests that conditions should always be set:

> Condition status values may be True, False, or Unknown. The absence of a condition should be interpreted the same as Unknown. 

I'll update this to instead provide custom reasons for healthy states of `Conflicted` and `Detached`.